applications/FluidDynamicsApplication/custom_utilities/fluid_test_utilities.h

```diff
@@ -0,0 +1,137 @@
+    ///@{
+
+    using NodeType = ModelPart::NodeType;
+
+    using PropertiesType = ModelPart::PropertiesType;
+
+    using DofsVectorType =  std::vector<Dof<double>::Pointer>;
+
+    using EquationIdVectorType = std::vector<std::size_t>;
+
+    using NodesContainerType = ModelPart::NodesContainerType;
+
+    using ConditionsContainerType = ModelPart::ConditionsContainerType;
+
+    using ElementsContainerType = ModelPart::ElementsContainerType;
+
+    ///@}
+    ///@name Static operations
+    ///@{
+
+    template<class TDataType>
+    static void AssignRandomValues(
+        TDataType& rValue,
+        const std::string& rSeed,
+        const int DomainSize,
+        const double MinValue = 0.0,
+        const double MaxValue = 1.0);
+
+    static ModelPart& CreateTestModelPart(
+        Model& rModel,
+        const std::string& rModelPartName,
+        const std::string& rElementName,
+        const std::string& rConditionName,
+        const std::function<void(PropertiesType&)>& rSetElementProperties,
+        const std::function<void(PropertiesType&)>& rSetConditionProperties,
+        const std::function<void(ModelPart&)>& rAddNodalSolutionStepVariablesFuncion,
+        const std::function<void(NodeType&)>& rAddDofsFunction,
+        const int BufferSize = 2);
+
+    template<class TDataType>
+    static void RandomFillHistoricalVariable(
+        ModelPart& rModelPart,
+        const Variable<TDataType>& rVariable,
+        const double MinValue = 0.0,
+        const double MaxValue = 1.0,
+        const int Step = 0);
+
+    template<class TContainerType, class TDataType>
+    static void RandomFillNonHistoricalVariable(
+        TContainerType& rContainer,
+        const Variable<TDataType>& rVariable,
+        const IndexType DomainSize,
+        const double MinValue = 0.0,
+        const double MaxValue = 1.0);
+
+    template<class TContainerType>
+    static void RunEntityGetDofListTest(
+        const TContainerType& rContainer,
+        const ProcessInfo& rProcessInfo,
+        const std::vector<const Variable<double>*>& rDofVariablesList);
+
+    template<class TContainerType>
+    static void RunEntityEquationIdVectorTest(
+        const TContainerType& rContainer,
+        const ProcessInfo& rProcessInfo,
+        const std::vector<const Variable<double>*>& rDofVariablesList);
+
+    template<class TContainerType>
+    static void RunEntityGetValuesVectorTest(
+        const TContainerType& rContainer,
+        const std::vector<const Variable<double>*>& rDofVariablesList);
+
+    template<class TContainerType>
+    static void RunEntityGetFirstDerivativesVectorTest(
+        const TContainerType& rContainer,
+        const std::function<Vector(const ModelPart::NodeType&)>& rValueRetrievalMethod);
+
+    template<class TContainerType>
+    static void RunEntityGetSecondDerivativesVectorTest(
+        const TContainerType& rContainer,
+        const std::function<Vector(const ModelPart::NodeType&)>& rValueRetrievalMethod);
```

Comment:
These methods will only work with the adjoint elements right?

Reply:
No, These should work with the any element.